//
// Copyright 2020 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package asylo;

// Various types used to test proto parsing utilities.

enum TestEnum {
  TEST_UNKNOWN = 0;
  TEST_VALUE_ONE = 1;
  TEST_VALUE_FORTY_TWO = 42;
}

message TestSubmessage {
  optional string string_field = 1;
}

message TestMessage {
  optional TestEnum enum_field = 1;
  optional uint32 int_field = 2;
  optional TestSubmessage message_field = 3;
}
